packages/headless/src/controllers/commerce/search-status/headless-search-status.ts

```diff
@@ -0,0 +1,19 @@
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {
+  SearchStatusState,
+  SearchStatus,
+} from '../../core/status/headless-core-status';
+import {buildCoreStatus} from '../core/status/headless-core-status';
+
+export type {SearchStatusState, SearchStatus};
+
+/**
+ * Creates a `SearchStatus` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `SearchStatus` controller instance.
+ * @internal
+ * */
+export function buildSearchStatus(engine: CommerceEngine): SearchStatus {
```

Comment:
I would propose to just use `search-summary` (both for search and product listing), which expose all this state/info already. It's a repetition.